data.py

```diff
@@ -212,21 +214,27 @@ class RefuelingTextFileConversion(Conversion):
     refuelings: RefuelingRepository
 
     def convert(self) -> None:
-        refuelings = self.refuelings_text_file.read()
-        for line in refuelings:
+        text = self.refuelings_text_file.read()
+        for line in text:
+            line = line[:-1]
             data = line.split('\t')
-            id = int(data[0])
-            date = datetime(data[1])
-            volume = float(data[2])
-            destination_id = int(data[3])
-            new_refueling = Refueling(id, date, volume, destination_id)
-            self.refuelings.add(new_refueling)
+            date = datetime.strptime(data[0], "%Y/%m/%d")
+            volume = float(data[1])
+            destination = next(destination for destination in destinations if destination.name == data[2])
+            new_refueling = Refueling(id=0, date=date, volume=volume, destination=destination)
+            refuelings.add(new_refueling)
+
 
+# TODO: not global!!
 
 destinations = DestinationRepository()
 refuelings = RefuelingRepository()
 trips = TripsRepository()
 text_file = TextFile("DESTINATIONS.txt")
 output = text_file.read()
-conversion = DestinationTextFileConversion(destinations=destinations)
-conversion.convert()
+destination_conversion = DestinationTextFileConversion(destinations=destinations)
+destination_conversion.convert()
+refueling_conversion = RefuelingTextFileConversion(refuelings)
+refueling_conversion.convert()
+trip_converstion = TripTextFileConversion(trips)
+trip_converstion.convert()
```

Comment:
Ok, so if I understand correctly: 
- `Conversion` class is a generic class to implement interface for other inherit classes like: `TripConversion`, `DestinationConversion` etc. 
- Every `ObjectConversion`class creates and returns that object from given list of strings.
- `DataStorage`  is a universal class to read data from .txt files for all type of objects:` Trips`, `Destination` etc. It returns a list of converted objects of given type
- Repository is a final class where date is saved in cache. 

`
class Repository(Generic[T]): # Generic instead of Repository
   # cache = None    Corret me if I'm wrong but in this case global cache will be 
   # equal for every instance, so we wouldn't be able to create separte 
   # Repositories for different objects
   def __init__(self, dataStorage: DataStorage[T]):
     self.dataStorage = dataStorage
     self.cache = None 
# In this case every instance of Repository class should have separate cache.`

Reply:
"# cache = None    Corret me if I'm wrong but in this case global cache will be"
Yes, you need to do it in `self.cache=...` way. Unlike other normal languages... :D

Reply:
> Repository is a final class where date is saved in cache.
It's not about cache. Repository should allow you to manipulate the objects you store: read, add, update, delete. Cache was just an idea that I got to not load the file every time `read` is called. It may be dangerous because you need to remember to invalidate that. Also, if some other process accesses the file, it may change it without your knowledge.